src/Compatibility/Microsoft.DotNet.PackageValidation/GetLastStablePackage.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using Microsoft.Build.Framework;
+using Microsoft.NET.Build.Tasks;
+using NuGet.Common;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace Microsoft.DotNet.PackageValidation
+{
+    public class GetLastStablePackage : TaskBase
+    {
+        [Required]
+        public string PackageId { get; set; }
+
+        public string PackageVersion { get; set; }
+
+        public string[] NugetFeeds { get; set; }
+
+        [Output]
+        public string LastStableVersion { get; set; }
+
+        protected override void ExecuteCore()
+        {
+            NuGetVersion currentPackageVersion = new NuGetVersion(PackageVersion);
+            NuGetVersion version = null;
+
+            foreach (string nugetFeed in NugetFeeds)
+            {
+                SourceRepository repository = Repository.Factory.GetCoreV3(nugetFeed);
+                FindPackageByIdResource resource = repository.GetResource<FindPackageByIdResource>();
+                SourceCacheContext cache = new SourceCacheContext();
+                IEnumerable<NuGetVersion> versions = resource.GetAllVersionsAsync(PackageId, cache, NullLogger.Instance, CancellationToken.None).Result;
```

Comment:
Does this always create a request or use some sort of caching? I think @ericstj had a similar comment before.